tutorials/get_started/tvmc_command_line_driver.py

```diff
@@ -0,0 +1,313 @@
+#
+# In the simplest form, tuning requires you to provide three things:
+#
+# - the target specification of the device you intend to run this model on;
+# - the path to an output file in which the tuning records will be stored, and finally,
+# - a path to the model to be tuned.
+#
+#
+# The example below demonstrates how that works in practice:
+#
+# .. code-block:: bash
+#
+#   tvmc tune \
+#     --target "llvm" \
+#     --output autotuner_records.json \
+#     resnet50-v2-7.onnx
+#
+#
+# Tuning sessions can take a long time, so ``tvmc tune`` offers many options to
+# customize your tuning process, in terms of number of repetitions (``--repeat`` and
+# ``--number``, for example), the tuning algorithm to be use, and so on.
+# Check ``tvmc tune --help`` for more information.
+#
+# As an output of the tuning process above, we obtained the tuning records stored
+# in ``autotuner_records.json``. This file can be used in two ways: as an input
+# to further tuning (via ``tvmc tune --tuning-records``), or as an input to the
+# compiler. The compiler will use the results to generate high performance code
+# for the model on your specified target.
+#
+# So now let's see how we can invoke the compiler using TVMC.
+#
+
+######################################################################
+# Compiling the model
+# -------------------
+#
+# To compile our model using TVM, we use ``tvmc compile``. The output we get from the
+# compilation process is a TAR package that can be used to run our model on the
+# target device.
+#
+# As previously mentioned, we can use tuning records in order to help the compilation
+# process to generate code that performs better for a given platform. To do that
+# we can use ``--tuning-records``. Check ``tvmc compile --help`` for more options.
+#
+# .. code-block:: bash
+#
+#   tvmc compile \
+#     --target "llvm" \
+#     --output compiled_module.tar \
+#     --tuning-records autotuner_records.json \
+#     resnet50-v2-7.onnx
+#
+# Once compilation finishes, the output ``compiled_module.tar`` will be created. This
+# can be directly loaded by your application and run via the TVM runtime APIs.
+# Alternatively, you can make use of the ``tvmc run`` command to quickly generate
+# predictions using the compiled module without having to write such an application.
+#
+
+
+######################################################################
+# Input pre-processing
+# --------------------
+#
+# In order to generate predictions, we will need two things:
+#
+# - the compiled module, which we just produced;
+# - a valid input to the model
+#
+# Each model is particular when it comes to expected tensor shapes, formats and data
+# types. For this reason, most models require some pre and
+# post processing, to ensure the input(s) is valid and to interpret the output(s).
+#
+# In TVMC, we adopted NumPy's ``.npz`` format for both input and output data.
+# This is a well-supported NumPy format to serialize multiple arrays into a file.
```

Comment:
Hmm it seems more straightforward to take an image directly. TVMC is a CLI so it seems weird to ask users to prepare another preprocessing script. The data serialization/deserialization overhead will also hurt the overall end-to-end performance. On the other hand, this can be done in the future.

Reply:
I agree with your point. Yes, there are plans to have such features in next versions. For ImageNet it it looks trivial enough, the challenge is that is should be flexible to allow users to bring their own pre/post processing. We'll certainly discuss more about that, so I'll take no action for now.